draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2407,6 +2400,26 @@ the subsection that gives the corresponding parameters.
 | secp256k1                 | {{suites-secp256k1}} |
 | BLS12-381                 | {{suites-bls12381}}  |
 
+## Implementing a hash-to-curve suite {#suites-howto}
+
+A hash-to-curve suite requires the following functions.
+Note that some of these require utility functions from {{utility}}.
+
+1. Working operations on the target elliptic curve (e.g., point addition,
+   scalar multiplication) and in the curve's base field (e.g., addition,
+   multiplication, square root).
+
+2. The hash-to-field function ({{hashtofield}}), including the expand\_message
+   variant ({{hashtofield-expand}}) and any constituent hash function or XOF.
+
+3. The mapping function specified by the suite; see {{mappings}}.
+
+4. A cofactor clearing function, which may be implemented as scalar multiplication
+   by h\_eff or as a faster equivalent method; see {{cofactor-clearing}}.
+
+5. The encoding function, either hash\_to\_curve or encode\_to\_curve; see
+   {{roadmap}}.
```

Comment:
Aiming for parallel structure:

```suggestion
1. Target elliptic curve operations, e.g., point addition and scalar multiplication.

2. Target elliptic curve base field operations, e.g., addition, multiplication, and square root.

3. The hash-to-field function; see {{hashtofield}}. This includes the expand\_message
   variant ({{hashtofield-expand}}) and any constituent hash function or XOF.

4. The suite-specific mapping function; see {{mappings}}.

5. A cofactor clearing function; see {{cofactor-clearing}}. This may be implemented as 
   scalar multiplication by h\_eff or as a faster equivalent method.

6. The desired encoding function; see {{roadmap}}. This is either hash\_to\_curve or 
   encode\_to\_curve.
```

(It may be over-generalizing a bit, but it seems we could probably define a hash-to-curve protocol as a tuple of these functions, e.g., (E, F, HashToField, MapToCurve, ClearCofactor, Encode).)